python/vowpalwabbit/pyvw.py

```diff
@@ -441,6 +441,8 @@ def from_example(self, ex):
 class simple_label(abstract_label):
     """Class for simple VW label"""
     def __init__(self, label=0., weight=1., initial=0., prediction=0.):
+        if not (isinstance(label, float) or isinstance(label, example)):
```

Comment:
How much does this check hurt performance?

Reply:
Performance in terms of coverage or time?

Reply:
runtime perf

Reply:
Can you please suggest some way to check it? Should it be checked with `%timeit`?

Reply:
With the check:
```
In [1]: %timeit pyvw.simple_label(1.0)
Out[1]: 1.6 µs ± 21.2 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```
Without the check:
```
In [1]: %timeit pyvw.simple_label(1.0)
Out[1]: 1.5 µs ± 9 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
this deserves more investigation, or a performance test with an e2e scenario

Reply:
@Smit-create have you looked into this?

Reply:
I want some help regarding this, how to do a performance test check. I have checked it using %timeit as above and it shows that after adding that check, it takes few microseconds more  than the current master

Reply:
@jackgerrits  So should I remove this check for now? as this PR looks fine after that. Also, I will open an issue related to [this](https://github.com/VowpalWabbit/vowpal_wabbit/pull/2276#issuecomment-590093274) to have a track of it.

Reply:
You could run the bindings on some dataset with and without the change and measure the difference in performanc

Reply:
I ran this on two cases 1. with the check and 2. without the check.
I ran this using `timeit` module on the (i) synthetic data set (ii) test_simple_label and measuring the time of run as well as importing the module.
Following is the data of 5 runs in each of the cases.
#### Case 1 : with the check
(i) On synthetic dataset i.e. `test_regressor_args()`
The time for 5 runs is:` [0.11250083799995991, 0.1103579630000695, 0.11039327899993623, 0.11107653700014453, 0.11065440400011539]`
(i) On `test_simple_label`
The time for 5 runs is: `[0.1577475270000832, 0.13949805800007198, 0.14299159899996994, 0.14037750700003926, 0.13951651699994727]`

#### Case 2: without the check
(i) On synthetic dataset i.e. `test_regressor_args()`
The time for 5 runs is: `[0.1079615889998422, 0.10585603900017304, 0.10556867700006478, 0.10575059799998598, 0.10852373700004136]`

(i) On `test_simple_label`
The time for 5 runs is:` [0.15763619900008052, 0.13817744999983006, 0.13808634600013647, 0.13846224700000676, 0.13818081999988863]`

Reply:
Okay based on those results, do you believe the checks are worth it?

Reply:
I think this should be added as it doesn't hit the performance too hard. You view on this?

Reply:
I agree

Reply:
Thanks!

Reply:
```suggestion
        if not isinstance(label, (example, float)):
```